py-polars/polars/datatypes/classes.py

```diff
@@ -564,7 +578,15 @@ def __hash__(self) -> int:
 
     def __repr__(self) -> str:
         class_name = self.__class__.__name__
-        return f"{class_name}(categories={self.categories!r})"
+        if len(categories := self.categories) <= 6:
+            categories = [",".join(f"{cat!r}" for cat in categories)]
+        else:
+            categories = (
+                [",".join(f"{cat!r}" for cat in categories[:3])]
+                + ["…"]
+                + [",".join(f"{cat!r}" for cat in categories[-3:])]
+            )
```

Comment:
This is probably a good idea, but it does break the 'guarantee' I mentioned here:
https://github.com/pola-rs/polars/issues/13152#issuecomment-1864455630

What's your take on this?

Reply:
Oof... I wouldn't (ever) tie `repr` to a guarantee about serialisability; there should be an API point for that 🤔 (Which could be pretty trivial in this case; a simple serialise/deserialise pair - in this case they could map to the `repr` where suitable, or to a non-truncated `repr` otherwise).

Reply:
I'm going to split this into two PRs - the type/uniqueness checks to ensure we're creating an Enum with valid categories, and the `repr` update, which may need to add a real serialisation API point 👌

Reply:
Done:
* #13356 
* #13357